packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,326 @@
+  set anchor(Anchor a) {
+    _anchor = a;
+    _onModifiedSizeOrAnchor();
   }
 
-  /// Set the top left position regardless of the anchor
-  set topLeftPosition(Vector2 position) {
-    this.position = position + (anchor.toVector2()..multiply(scaledSize));
+  /// The logical size of the component. The game assumes that this is the
+  /// approximate size of the object that will be drawn on the screen.
+  /// This size will therefore be used for collision detection and tap
+  /// handling.
+  ///
+  /// This property can be reassigned at runtime, although this is not
+  /// recommended. Instead, in order to make the [PositionComponent] larger
+  /// or smaller, change its [scale].
+  final NotifyingVector2 _size;
+  Vector2 get size => _size;
+  set size(Vector2 size) => _size.setFrom(size);
+
+  /// The width of the component in local coordinates. Note that the object
+  /// may visually appear larger or smaller due to application of [scale].
+  double get width => _size.x;
+  set width(double w) => _size.x = w;
+
+  /// The height of the component in local coordinates. Note that the object
+  /// may visually appear larger or smaller due to application of [scale].
+  double get height => _size.y;
+  set height(double h) => _size.y = h;
+
+  /// The "physical" size of the component. This is the size of the
+  /// component as seen from the parent's perspective, and it is equal to
+  /// [size] * [scale]. This is a computed property and cannot be
+  /// modified by the user.
+  final Vector2 _scaledSize;
+  Vector2 get scaledSize => _scaledSize;
+
+  /// Measure the distance (in parent's coordinate space) between this
+  /// component's anchor and the [other] component's anchor.
+  double distance(PositionComponent other) =>
+      position.distanceTo(other.position);
+
+  //----------------------------------------------------------------------------
+  // Coordinate transformations
+  //----------------------------------------------------------------------------
+
+  /// Test whether the `point` (given in global coordinates) lies within this
+  /// component. The top and the left borders of the component are inclusive,
+  /// while the bottom and the right borders are exclusive.
+  @override
+  bool containsPoint(Vector2 point) {
+    final local = absoluteToLocal(point);
+    return (local.x >= 0) &&
+        (local.y >= 0) &&
+        (local.x < _size.x) &&
+        (local.y < _size.y);
   }
 
-  /// Get the absolute top left position regardless of whether it is a child or not
-  Vector2 get absoluteTopLeftPosition {
-    final p = parent;
-    if (p is PositionComponent) {
-      return p.absoluteTopLeftPosition + topLeftPosition;
-    } else {
-      return topLeftPosition;
+  /// Convert local coordinates of a point [p] inside the component into
+  /// the parent's coordinate space. Point [p] can be either a [Vector2],
+  /// an [Anchor] (relative to component's [size]), or an [Offset].
+  Vector2 positionOf(dynamic p) {
```

Comment:
I wonder if we'd maybe rather have this as two methods, so that the type system can catch errors before runtime.

Reply:
Good point. What should they be named? And should we care about `Offset`s at all?

Reply:
No, `Offset` can be skipped. We're trying to keep everything possible as `Vector2`.
A name suggestion could be to keep this as `positionOf` and introduce `positionOfAnchor` (also check out the `Anchor` method that does conversion to a position).

Reply:
done